conan/tools/cmake/toolchain.py

```diff
@@ -187,11 +187,9 @@ def context(self):
                    }.get(libcxx)
             if lib:
                 lib = "-library={}".format(lib)
-        elif compiler == "gcc":
-            # we might want to remove this "1", it is the default in most distros
-            if libcxx == "libstdc++11":
-                glib = "1"
-            elif libcxx == "libstdc++":
+
+        if compiler in ['clang', 'apple-clang', 'gcc']:
+            if libcxx == "libstdc++":
```

Comment:
Lets be aware that from https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html

```
Using the default configuration options for GCC the default value of the macro is 1 which causes the new ABI to be active, so to use the old ABI you must explicitly define the macro to 0 before including any library headers. (Be aware that some GNU/Linux distributions configure GCC 5 differently so that the default value of the macro is 0 and users must define it to 1 to enable the new ABI.)
```
So if we go this way, users from those distros will need to customize the ``CMakeToolchain``, something like this:

```python
def generate(self):
    tc = CMakeToolchain(self)
    tc.blocks["libcxx"].values["glibcxx"] = "1" if self.settings.libcxx == "libstdc++11" else "0"
    tc.generate()
```

There were requests to eliminate the default ``GLIBCXX_USE_CXX11_ABI=1``, as it was raising some warnings (undefined in some compilers, but not errors), so it seems there is no perfect solution for 100% of users.